core/src/main/scala/kafka/server/BrokerLifecycleManager.scala

```diff
@@ -431,8 +431,18 @@ class BrokerLifecycleManager(val config: KafkaConfig,
     }
 
     override def onTimeout(): Unit = {
-      info("Unable to send a heartbeat because the RPC got timed out before it could be sent.")
-      scheduleNextCommunicationAfterFailure()
+      if (_state == BrokerState.PENDING_CONTROLLED_SHUTDOWN) {
```

Comment:
I think one way we can fix this test is to let `KafkaBroker.shutdown` accept a timeout parameter. Possibly we could overload it like this:
```
def shutdown(timeoutMs: Long)
def shutdown() = shutdown(5000)
```
Then we can change the test to just wait a couple seconds. What do you think?

Reply:
If we modify the interface in this way it looks like the timeout refers to the overall shutdown operation, but the timeout is specific to the controlled shutdown attempt.

Since the test applies only to KRaft we can cast the broker to the known type (`BrokerServer`) and modify a new _VisibleForTesting_ variable that holds the default timeout. WDYT?